core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3801,6 +3803,107 @@ private ExpressionNode rewriteAggregate(ExpressionNode agg, QueryModel model) {
         return agg;
     }
 
+    /**
+     Rewrites
+     SELECT LAST(timestamp) FROM table_name;
+     AND
+     SELECT max(timestamp) FROM table_name;
+     into query which can is optimised search
+     SELECT timestamp from (SELECT timestamp from table_name order by timestamp) limit 1
+     This enables FrameBackwardScan for aggregation queries containing LAST keyword
+     AND
+     Rewrites
+     SELECT FIRST(timestamp) FROM table_name;
+     AND
+     SELECT min(timestamp) FROM table_name;
+     into query which can is optimised search
+     SELECT timestamp from (SELECT timestamp from table_name) limit 1
+     This disables invoking group by workers
+     */
+    private void optimiseAggregateFunctions(QueryModel parent){
```

Comment:
Formatting is still incorrect. If you're using IDEA, please run built-in formatter for the changes
```suggestion
    private void optimiseAggregateFunctions(QueryModel parent) {
```